src/providers/aws.rs

```diff
@@ -1321,6 +1304,31 @@ impl RegionLauncher {
             .instrument(key_span)
             .await;
         }
+
+        if !self.security_group_id.trim().is_empty() {
+            let group_span = tracing::trace_span!("security_group", id = %self.security_group_id);
+            async {
+                tracing::trace!("removing security group.");
+                // clean up security groups and keys
+                loop {
```

Comment:
We probably want a limit on how long we retry this for. Like a minute or something?